galaxy_importer/ansible_test/builders/pulp_build.py

```diff
@@ -15,7 +15,271 @@
+                url=f'{self.api_url}{task_href}',
+                auth=self.basic_auth)
+            status = r.json()['state']
+            if status == 'failed' or status == 'canceled':
+                self.cleanup()
+                raise exceptions.AnsibleTestError(
+                    f'Could not create ContainerDistribution: \
+                        {r.status_code} {r.reason} {r.text}')
+            elif status == 'completed':
+                return r.json()['created_resources'][0]
+            time.sleep(API_CHECK_DELAY_SECONDS)
+
+    def _get_registry_href(self):
+        r = requests.get(
+            url=f'{self.api_url}{self.container_distribution_href}',
+            auth=self.basic_auth,
+        )
+        if r.status_code == 200:
+            self._add_registry_to_conf(r.json()['registry_path'])
+            return r.json()['registry_path']
+        else:
+            raise exceptions.AnsibleTestError(
+                f'Could not retrieve Registry href: \
+                    {r.status_code} {r.reason} {r.text}')
+
+    def _add_registry_to_conf(self, registry_href):
+        reg_file = f'/home/{os.getlogin()}/.config/containers/registries.conf'
+        if not os.path.exists(reg_file):
+            path = f'home/{os.getlogin()}/.config/containers'.split('/')
+            for index, item in enumerate(path):
+                current_path = Build._get_current_path(index, path)
+                if not os.path.exists(current_path):
+                    subprocess.run(['mkdir', f'{current_path}'])
+                else:
+                    continue
+            subprocess.run(['touch', f'{reg_file}'])
+
+        with open(reg_file, 'r+') as f:
+            if os.stat(reg_file).st_size == 0:
+                c = '[registries.insecure]\n'
+                c = c + f'registries = ["{registry_href}"]'
+            else:
+                c = f.readlines()
+                for index, line in enumerate(c):
+                    if 'registries = [' in line and registry_href not in line:
+                        c[index] = line.replace(
+                            'registries = [',
+                            f"registries = ['{registry_href}', ")
+            f.seek(0)
+            f.truncate()
+            f.writelines(c)
+
+    @staticmethod
+    def _get_current_path(index, path):
+        d = ''
+        i = 0
+        while i <= index:
+            d = d + f'/{path[i]}'
+            i = i + 1
+        return d
+
+    def cleanup(self):
+        """Clean up temporary data structures and files"""
+        self._delete_container_repository(self.container_repository_href)
+        self._delete_container_distribution(self.container_distribution_href)
+        self._delete_artifact(self.dockerfile_artifact_href)
+        self._delete_artifact(self.collection_artifact_href)
+        os.remove(self.dockerfile)
+        os.remove(self.collection_file)
+
+    def _delete_container_repository(self, repository_href):
```

Comment:
may be able to combine all 3 `_delete_*()` methods, passing in the url, checking for an "ok" status code via `requests.codes.ok` and in the exception saying which url caused the error - I am assuming that gives some context on what is attempting to be deleted.